src/App.tsx

```diff
@@ -15,36 +16,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type Params = {
+  sortfield: SortType;
+  isReversed: boolean;
+};
+
+enum SortType {
+  default = '',
+  alphabet = 'alphabet',
+  length = 'length',
+}
+
+function prepareGoods(goods: string[], params: Params): string[] {
+  const { sortfield, isReversed } = params;
+
+  let preparedGoods = [...goods].sort((good1, good2) => {
+    switch (sortfield) {
+      case SortType.alphabet:
+        return good1.localeCompare(good2);
+      case SortType.length:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    preparedGoods = [...preparedGoods].reverse();
```

Comment:
```suggestion
    preparedGoods.reverse();
```
You have already create a copy, you don't need to do it one more time